nav2_util/include/nav2_util/service_client.hpp

```diff
@@ -126,6 +139,41 @@ class ServiceClient
     return response.get();
   }
 
+  /**
+  * @brief Invoke the service and block until completed
+  * @param request The request object to call the service using
+  * @return std::shared_future<typename ResponseType::SharedPtr> The shared future of the service response
+  */
+  std::shared_future<typename ResponseType::SharedPtr> invoke_shared(
```

Comment:
What's the value of this if we can return the response instead? Why return the shared future?